### PR TITLE
Fix/bounced event flow

### PR DESCRIPTION
**Pull request by dannil76**
## Explain the changes you’ve made
The submittCompletion fired on occasions when it should not. This is due to that the status ("active:submitted") is set on several occasions throughout a case's life-span, but the completion should only be sent once when state is "VIVA_COMPLETION_REQUIRED" and the status is "active:submitted".

### CODE REVIEW

**Comment by sebastianthulin** [comment]
Quality gate Quality gate: **_Passed_** 
 


            
 No issues Everything looks good
 [Details](https://app.embold.io/organization/gh/helsingborg-stad/repositories/602c13c73b93a7537d767e421a43afaa/pull-request/274)

**Comment by helsingborgsstad** [comment]
Task linked: [CU-1m3n75d Bugg: dokument för stickprov laddas upp dubbelt i Viva](https://app.clickup.com/t/1m3n75d)

File: services/viva-ms/serverless.yml [hunk 1]
```diff
@@ -160,20 +160,13 @@ functions:
...
-                  provider:
-                    S: ["VIVA"]
                   status:
-                    M:
-                      type:
-                        S: [{ "prefix": "active:submitted" }]
-                  state:
-                    S: ["VIVA_COMPLETION_REQUIRED"]
+                type: [{ "prefix": "active:submitted" }]
+              state: ["VIVA_COMPLETION_REQUIRED"]
```
Comment (TeddyAndersson) [inline services/viva-ms/serverless.yml:169]: Fix Indentation to keep serverless happy

File: services/viva-ms/lambdas/submitCompletion.js [hunk 2]
```diff
@@ -1,19 +1,34 @@
...
 import S3 from '../../../libs/S3';
 import { VIVA_COMPLETION_RECEIVED } from '../../../libs/constants';
 
 import vivaAdapter from '../helpers/vivaAdapterRequestClient';
 
 export async function main(event, context) {
-  const caseItem = parseDynamoDBItemFromEvent(event);
+  const { caseKeys } = event.detail;
+
+  const [getCaseItemError, { Item: caseItem }] = await getStoredUserCase(
```
Comment (TeddyAndersson) [inline services/viva-ms/lambdas/submitCompletion.js:18]: There are no checks to see if the returned Item is empty. IE what happens if the query is successful but the data returned is empty.

**Review by TeddyAndersson — CHANGES_REQUESTED** [review]

**Review by MasterKenth — APPROVED** [review]

**Comment by MasterKenth** [comment]
I haven't tested this but it seems fine.

**Review by TeddyAndersson — COMMENTED** [review]
Ship it! 🚀

**Pull request merged**